src/main/java/org/folio/rest/impl/PatronServicesResourceImpl.java

```diff
@@ -1167,4 +1177,42 @@ private String buildQueryWithRequesterId(String requesterId, String sortBy) {
     return format("(requesterId==%s and status==Open*)", requesterId);
   }
 
+  private CompletableFuture<Boolean> isEcsTlrFeatureEnabled(VertxOkapiHttpClient httpClient,
+    Map<String, String> okapiHeaders) {
+
+    return httpClient.get(ECS_TLR_SETTINGS_URL_PATH.getValue(), Map.of(), okapiHeaders)
+      .thenApply(ResponseInterpreter::extractResponseBody)
+      .thenCompose(body -> getEcsTlrFeatureValue(body, httpClient, okapiHeaders));
+  }
+
+  private CompletableFuture<Boolean> getEcsTlrFeatureValue(JsonObject body,
+    VertxOkapiHttpClient client, Map<String, String> okapiHeaders) {
+
+    return Objects.nonNull(body)
```

Comment:
Body can be non-null even if response has non-2xx response status

Reply:
fixed